crates/rolldown_binding/src/types/binding_outputs.rs

```diff
@@ -96,7 +99,10 @@ impl FinalBindingOutputs {
       let mut inner = inner.lock().expect("PoisonError raised");
       inner.iter_mut().for_each(|o| match o {
         rolldown_common::Output::Asset(asset) => {
-          assets.push(BindingOutputAsset::new(unsafe { std::mem::transmute(asset.as_mut()) }));
+          assets.push(BindingOutputAsset::new(unsafe {
+            &mut *(std::ptr::from_mut::<std::sync::Arc<rolldown_common::EmittedAsset>>(asset)
+              .cast::<rolldown_common::EmittedAsset>())
+          }));
```

Comment:
I don't thnk this is a safe cast.

Reply:
Yes, I am currently investigating this issue. I'm not sure if we should really avoid cloning, it seems to affect the `files` in `FileEmitter`.

Reply:
According to the benchmark `asset` clone is not the bottleneck, clone it is fine.